types/twine-sugarcube/extensions.d.ts

```diff
@@ -6,6 +6,111 @@ declare global {
      * @param array The array being processed.
      */
     type ArrayPredicate<T, ThisType> = (this: ThisType, value: T, index: number, array: T[]) => boolean;
+
+    interface ReadonlyArray<T> {
+        /**
+         * Returns the number of times that the given member was found within the array, starting the search at position.
+         * @param needle The member to count.
+         * @param position The zero-based index at which to begin searching for needle. If omitted, will default to 0.
+         * @since SugarCube 2.0.0
+         * @example
+         * // Given: $fruits = ["Apples", "Oranges", "Plums", "Oranges"]
+         * $fruits.count("Oranges")     → Returns 2
+         * $fruits.count("Oranges", 2)  → Returns 1
+         */
+        count(needle: T, position?: number): number;
+
+        /**
+         * Returns the number of times that members within the array pass the test implemented by the given predicate function.
+         * @param predicate The function used to test each member. It is called with three arguments:
+         * value: The member being processed.
+         * index: (optional, integer) The index of member being processed.
+         * array: (optional, array) The array being processed.
+         * @param thisArg The value to use as this when executing predicate.
+         * @since SugarCube 2.36.0
+         * @example
+         * // Given: $fruits = ["Apples", "Oranges", "Plums", "Oranges"]
+         * $fruits.countWith(function (fruit) { return fruit === "Oranges"; })  → Returns 2
+         */
+        countWith<PredicateThisArg>(predicate: ArrayPredicate<T, PredicateThisArg>, thisArg: PredicateThisArg): number;
+        countWith(predicate: ArrayPredicate<T, undefined>): number;
+
+        /**
+         * Returns the first member from the array. Does not modify the original.
+         * @since SugarCube 2.2.7.0
+         * @example
+         * // Given: $pies = ["Blueberry", "Cherry", "Cream", "Pecan", "Pumpkin"]
+         * $pies.first() // Returns "Blueberry"
+         */
+        first(): T | undefined;
+
+        /**
+         * Returns a new array consisting of the flattened source array (i.e. flat map reduce). Does not modify the original.
```

Comment:
There should be a deprecation note/tag here.  E.g.,

```
* [DEPRECATED] Returns a new …
```

It doesn't have to be exactly that, though that's generally how I do it in the source code—e.g., [extensions.js#L1875](https://github.com/tmedwards/sugarcube-2/blob/27903a854b7bedbb9fa60efa33e47c99bab7a359/src/lib/extensions.js#L1875)—but I do want deprecated APIs marked as such for inline editor hints.

Reply:
Done.